### ISSUE TITLE
Make an object-oriented interface for embedding models

### ISSUE THREAD

**Issue by jtcramer** [marl/openl3#22]
An object interface for extracting embeddings would make it simpler to extract multiple embeddings and would allow for the embedding model to just be loaded once instead of every time the `get_embedding` function is called.

At least to begin with, the interface could just be something like

```
m = EmbeddingModel(input_repr="mel256", content_type="music", 
                                    embedding_size=6144, center=True, hop_size=0.1):
emb, ts = m.get_embedding(audio, sr, verbose=1)
m.process_file(filepath, output_dir=None, suffix=None)
```

Basically, we'd just put all of the functions from https://github.com/marl/openl3/blob/master/openl3/core.py into a class.

**Comment by justinsalamon** [comment]
Sounds reasonable. Perhaps I'd move the `hop_size` and `embedding_size` params into the `get_embedding` function call, as these are independent of the specific model file loaded?

**Comment by justinsalamon** [comment]
Do we still want to do this now that you can pass a model to `get_embedding`?

**Comment by jtcramer** [comment]
I think it might still be a decent idea, but as of now there's no rush towards an OOP model. The main benefit I suppose would be so that you don't have to pass around a model object. But on the other hand it's not really that much bookkeeping. And perhaps it would be best to err on the side of simplicity before overengineering an OOP API?

**Comment by justinsalamon** [comment]
Yeah could still be handy, let’s keep things stable for now and we can revisit this question down the line if/when we get more iser feedback
